### PR TITLE
br/pkg/rtree: migrate test-infra to testify for br/pkg/rtree

### LINKED ISSUE #28178: migrate test-infra to testify for `br/pkg/rtree`

### PR DESCRIPTION
**Pull request by Defined2014**
### What problem does this PR solve?

Issue Number: close #28178 

Problem Summary: migrate test-infra to testify for br/pkg/rtree

### What is changed and how it works?

What's Changed: migrate test-infra to testify for br/pkg/rtree

### Release note 

```release-note
None
```

### CODE REVIEW

File: br/pkg/rtree/rtree_test.go [hunk 1]
```diff
@@ -85,92 +78,93 @@ func (s *testRangeTreeSuite) TestRangeTree(c *C) {
...
-func (s *testRangeTreeSuite) TestRangeIntersect(c *C) {
+func TestRangeIntersect(t *testing.T) {
+	t.Parallel()
 	rg := newRange([]byte("a"), []byte("c"))
 
 	start, end, isIntersect := rg.Intersect([]byte(""), []byte(""))
-	c.Assert(isIntersect, Equals, true)
-	c.Assert(start, DeepEquals, []byte("a"))
-	c.Assert(end, DeepEquals, []byte("c"))
+	require.Equal(t, true, isIntersect)
```
Comment (kennytm) [inline br/pkg/rtree/rtree_test.go]:
```suggestion
	require.True(t, isIntersect)
```

File: br/pkg/rtree/rtree_test.go [hunk 2]
```diff
@@ -38,11 +31,11 @@ func (s *testRangeTreeSuite) TestRangeTree(c *C) {
...
 		incomplete := rangeTree.GetIncompleteRange(startKey, endKey)
-		c.Logf("%#v %#v\n%#v\n%#v\n", startKey, endKey, incomplete, ranges)
-		c.Assert(len(incomplete), Equals, len(ranges))
+		t.Logf("%#v %#v\n%#v\n%#v\n", startKey, endKey, incomplete, ranges)
+		require.Equal(t, len(ranges), len(incomplete))
 		for idx, rg := range incomplete {
-			c.Assert(rg.StartKey, DeepEquals, ranges[idx].StartKey)
-			c.Assert(rg.EndKey, DeepEquals, ranges[idx].EndKey)
+			require.Equal(t, ranges[idx].StartKey, rg.StartKey)
+			require.Equal(t, ranges[idx].EndKey, rg.EndKey)
```
Comment (kennytm) [inline br/pkg/rtree/rtree_test.go:38]:
```suggestion
			require.Equalf(t, ranges[idx].StartKey, rg.StartKey, "idx=%d", idx)
			require.Equalf(t, ranges[idx].EndKey, rg.EndKey, "idx=%d", idx)
```

File: br/pkg/rtree/rtree_test.go [hunk 3]
```diff
@@ -85,92 +78,93 @@ func (s *testRangeTreeSuite) TestRangeTree(c *C) {
...
-	c.Assert(end, DeepEquals, []byte("c"))
+	require.Equal(t, true, isIntersect)
+	require.Equal(t, []byte("a"), start)
+	require.Equal(t, []byte("c"), end)
 
 	start, end, isIntersect = rg.Intersect([]byte(""), []byte("a"))
-	c.Assert(isIntersect, Equals, false)
-	c.Assert(start, DeepEquals, []byte(nil))
-	c.Assert(end, DeepEquals, []byte(nil))
+	require.Equal(t, false, isIntersect)
```
Comment (kennytm) [inline br/pkg/rtree/rtree_test.go]:
```suggestion
	require.False(t, isIntersect)
```

File: br/pkg/rtree/rtree_test.go [hunk 4]
```diff
@@ -85,92 +78,93 @@ func (s *testRangeTreeSuite) TestRangeTree(c *C) {
...
+	require.Equal(t, []byte(nil), start)
+	require.Equal(t, []byte(nil), end)
 
 	start, end, isIntersect = rg.Intersect([]byte(""), []byte("b"))
-	c.Assert(isIntersect, Equals, true)
-	c.Assert(start, DeepEquals, []byte("a"))
-	c.Assert(end, DeepEquals, []byte("b"))
+	require.Equal(t, true, isIntersect)
```
Comment (kennytm) [inline br/pkg/rtree/rtree_test.go]:
```suggestion
	require.True(t, isIntersect)
```

and so on for the remaining 5.

**Comment by Defined2014** [comment]
/cc @tisonkun 
PTAL

**Review by kennytm — COMMENTED** [review]
rest LGTM

**Review by kennytm — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
Thanks for your contribution! LGTM.

**Pull request merged**